llvm/lib/CodeGen/AsmPrinter/DwarfDebug.cpp

```diff
@@ -3518,11 +3525,16 @@ void DwarfDebug::addDwarfTypeUnitType(DwarfCompileUnit &CU,
       return;
     }
 
-    // If the type wasn't dependent on fission addresses, finish adding the type
-    // and all its dependent types.
     for (auto &TU : TypeUnitsToAdd) {
       InfoHolder.computeSizeAndOffsetsForUnit(TU.first.get());
       InfoHolder.emitUnit(TU.first.get(), useSplitDwarf());
+      if (getDwarfVersion() >= 5 &&
+          getAccelTableKind() == AccelTableKind::Dwarf) {
+        addTypeUnitSymbol(*TU.first);
+        AccelTypeUntsDebugNames.convertDieToOffset();
+        AccelDebugNames.addTypeEntries(AccelTypeUntsDebugNames);
+        AccelTypeUntsDebugNames.clear();
```

Comment:
This code doesn't look quite right - there might be multiple type units being added here (hence the loop over `TypeUnitsToAdd`) - and this would add all the names once the first type unit has been laid out - but before other type units have been laid out, so I would guess that'd cause `convertDieToOffset` to fail. 

This convert/add/clear part should probably go after the `TypeUnitsToAdd` loop, I think? (the `addTypeUnitSymbol` probably still goes inside the loop)

Reply:
yes, good catch. thank you.